yarn-project/circuits.js/src/structs/max_block_number.ts

```diff
@@ -0,0 +1,80 @@
+import { Fr } from '@aztec/foundation/fields';
+import { BufferReader, FieldReader, serializeToBuffer, serializeToFields } from '@aztec/foundation/serialize';
+import { FieldsOf } from '@aztec/foundation/types';
+
+import { MAX_BLOCK_NUMBER_LENGTH } from '../constants.gen.js';
+
+/**
+ * Maximum block number.
+ */
+export class MaxBlockNumber {
+  constructor(
+    /**
+     * Whether a max block number was requested.
+     */
+    public isSome: boolean,
+    /**
+     * The requested max block number, if isSome is true.
+     */
+    public value: Fr,
```

Comment:
Should this one be a number to now run into the case where it is larger than the `u32`?

Reply:
We seem to be using `number` for `u32` (e.g. `public readU32(): number` in `FieldReader`), but this is wrong: `number` is a floating point value, we should use `bigint` or `Fr` instead. This is a larger issue/refactor however.